### PR TITLE
Adjust docbuild for moved src

### PR DESCRIPTION
**Pull request by mwichmann**
`src/engine/SCons` moved to `SCons`, affects wired in paths in documentation, and doc building scripts.

At this stage of the PR, changes wired-in paths like `"../../doc..."`, and the places where the scripts in `bin` build paths.  Also includes regenerated files in `doc/generated`, including examples.

The three scripts in bin are all now working as expected.

If you actually run scons, results go to `build`, which otherwise isn't being used now.  This includes copying part of the tree to build - the docbook, gs and zip tools. Is this okay?

Signed-off-by: Mats Wichmann <mats@linux.com>

## Contributor Checklist:

* [ ] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [ ] I have updated `CHANGES.txt` (and read the `README.rst`)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: doc/SConscript [hunk 1]
```diff
@@ -505,7 +505,7 @@ if not skip_doc:
         # XXX Should be in common with reading the same thing in
         # the SConstruct file.
         # bootstrap.py runs outside of SCons, so need to process the path
-        e = Dir(os.path.join('#src', 'engine')).rstr()
+        e = Dir('#').rstr()
         sources = bootstrap.parseManifestLines(e, os.path.join(e, 'MANIFEST.in'))
```
Comment (bdbaddog) [inline doc/SConscript:509]: I think this is not going to work against the current MANIFEST.in file, but could be wrong.

File: doc/generated/examples/EnumVariable_map_1.xml [hunk 2]
```diff
@@ -1,4 +1,3 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<screen xmlns="http://www.scons.org/dbxsd/v1.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.scons.org/dbxsd/v1.0 http://www.scons.org/dbxsd/v1.0/scons.xsd">% <userinput>scons -Q COLOR=navy foo.o</userinput>
+<screen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.scons.org/dbxsd/v1.0" xsi:schemaLocation="http://www.scons.org/dbxsd/v1.0 http://www.scons.org/dbxsd/v1.0/scons.xsd">% <userinput>scons -Q COLOR=navy foo.o</userinput>
```
Comment (bdbaddog) [inline doc/generated/examples/EnumVariable_map_1.xml:1]: Is dropping these `<?xml's >`important?

File: doc/generated/examples/environments_missing2_1.xml [hunk 3]
```diff
@@ -1,6 +1,5 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<screen xmlns="http://www.scons.org/dbxsd/v1.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.scons.org/dbxsd/v1.0 http://www.scons.org/dbxsd/v1.0/scons.xsd">% <userinput>scons -Q</userinput>
+<screen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.scons.org/dbxsd/v1.0" xsi:schemaLocation="http://www.scons.org/dbxsd/v1.0 http://www.scons.org/dbxsd/v1.0/scons.xsd">% <userinput>scons -Q</userinput>
 
-scons: *** NameError `MISSING' trying to evaluate `$MISSING'
+scons: *** NameError `name 'MISSING' is not defined' trying to evaluate `$MISSING'
```
Comment (bdbaddog) [inline doc/generated/examples/environments_missing2_1.xml:3]: Just a note for myself to check when this code change happened?

File: doc/SConscript (hunk 1 above)
Comment (mwichmann) [reply doc/SConscript:509]: You'll have to tell me... I just dropped src and engine, and after that there was nothing left to "join", there was no other thought behind this.

File: doc/generated/examples/EnumVariable_map_1.xml (hunk 2 above)
Comment (mwichmann) [reply doc/generated/examples/EnumVariable_map_1.xml:1]: yeah, lxml won't produce it, as it considers the presence of an encoding declaration here to be a fatal error.

Comment (mwichmann) [reply doc/generated/examples/EnumVariable_map_1.xml:1]: https://lxml.de/FAQ.html#why-can-t-lxml-parse-my-xml-from-unicode-strings

**Pull request merged**